reconcile/aws_cloudwatch_log_retention/integration.py

```diff
@@ -24,7 +25,7 @@
 class AWSCloudwatchLogRetention(BaseModel):
     name: str
     acct_uid: str
-    log_regex: str
+    log_regex: typing.Pattern
```

Comment:
Also need to change init to use `re.compile` in https://github.com/app-sre/qontract-reconcile/pull/3739/files#diff-773495110cdde5ed044f7f1bbf0eb68bf7ba5d387c6e3985cedaea64ec613760R43

```python
log_regex=re.compile(aws_acct_field["regex"]),
```